pkg/network/cache/client.go

```diff
@@ -37,10 +41,10 @@ func (c *ClientCache) Get(netAddr network.AddressGroup) (client.Client, error) {
 	//  but network.StringifyGroup is order-dependent.
 	//  This works until the same mixed group is transmitted
 	//  (for a network map, it seems to be true).
-	mAddr := network.StringifyGroup(netAddr)
+	cacheKey := hex.EncodeToString(info.PublicKey()) + network.StringifyGroup(netAddr)
```

Comment:
I guess we can drop `netAddr` part from `cacheKey` completely, because public keys are good enough for node identification.

Reply:
For node identification - agree. But if storage node changes the network endpoints in network map, cache will still return cached clients with old endpoints. I guess we need to think it over again.